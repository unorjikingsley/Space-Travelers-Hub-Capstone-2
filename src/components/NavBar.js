import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
  const links = [
    {
      name: 'Rockets',
      path: '/',
    },
    {
      name: 'Missions',
      path: 'Missions',
    },
    {
      name: 'My Profile',
      path: 'Profile',
    },
  ];

  return (
    <nav>
      <h1>Space Travellers&apos; Hub</h1>
      <ul>
        {links.map((link) => (
          <li key={link.name}>
            <NavLink
              to={link.path}
            >
              {link.name}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavBar;
