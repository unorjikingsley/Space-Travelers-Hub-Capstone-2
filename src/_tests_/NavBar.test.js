import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

it('renders correctly', () => {
  const navBar = renderer
    .create(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    )
    .toJSON();
  expect(navBar).toMatchSnapshot();
});
