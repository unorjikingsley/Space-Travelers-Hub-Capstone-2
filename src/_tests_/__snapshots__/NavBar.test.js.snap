// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo"
  >
    <img
      alt="space traveler's hub logo"
      className="logo-img"
      src="logo-space-travelers-inc.png"
    />
    <h1
      className="head"
    >
      Space Traveller's Hub
    </h1>
  </div>
  <ul
    className="nav-list"
  >
    <li
      className="list"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="list"
    >
      <a
        className="unactive"
        href="/Missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="list"
    >
      <a
        className="unactive"
        href="/Profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
